### DISCUSSION TITLE
Multiple loggers having an empty string as name, OK?

### DISCUSSION THREAD

**Discussion by N-Dekker** [gabime/spdlog#2639] [Q&A] [answered]
Is it OK to have multiple loggers (`spdlog::logger` objects) in an application that all have an empty string `""` as name (as specified by the `name` parameter of the constructors of `spdlog::logger`)?  I'm asking because it seems to me that [our application](https://github.com/SuperElastix/elastix) does not really need those loggers to be named.

**Comment by tt4g** [comment, marked as answer]
`default_logger` uses an empty string for its name.

https://github.com/gabime/spdlog/blob/927cc29444a294d76e83dfb898e797dc431ce094/include/spdlog/details/registry-inl.h#L37-L49

Loggers with the same name cannot be registered in the global registry.

https://github.com/gabime/spdlog/blob/927cc29444a294d76e83dfb898e797dc431ce094/include/spdlog/details/registry-inl.h#L298-L303

You can avoid name conflicts by replacing the default logger with `spdlog::set_default_logger()`.

**Reply by N-Dekker** [reply]
Thanks @tt4g I did not yet try the global registry, but I see what you mean!
